flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasK.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param;
+
+import org.apache.flink.ml.param.IntParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.WithParams;
+
+/** Interface for the shared K param. */
+public interface HasK<T> extends WithParams<T> {
```

Comment:
The definition of this parameter is pretty much the same as the existing `KMeansModelParams::K`. Could we update `KMeansModelParams` to use `HasK`?

We can update the Java doc to be something like `The number of clusters or neighbors of the algorithm`. And if the default value of `KNN` can not be same as that of `KMeans` (which has `k=2`), we can update one of the algorithm to override the default value of this parameter in the estimator/model constructor.

What do you think?

Reply:
I think this two params can't use one param. for :
1. default value different.
2. description different.
3. ParamValidators different. 

If update one with the other, we need update so many different info. Use two different params maybe better.

Reply:
@lindong28 Shall we keep `K` as an inner interface for Knn here? Similar as KMeans did now.

Many machine learning algs contains a parameter `k`, but they share different semantics. Combining all of them seems a bit over-design to me.

Reply:
@zhipeng93 @weibozhao If this parameter does not plan to be shared by multiple algorithms, could it be defined inside `KnnModelParams` (similar to how we define `k` in `KMeansModelParams`) instead being defined in `HasK`?

If this parameter does intend to be shared by multiple algorithms, in addition to putting it in `HasK`, it is probably useful to use a class name that clearly specifies how it is going to be different from the `KMeansModelParams::K`. Note tha we may want to make `KMeansModelParams::K` a sharable parameter in the future as well. We will be good to think ahead and avoid ambiguous names.

Reply:
OK, I have defined this params inside KnnModelParams already.